src/components/AppIndex.vue

```diff
@@ -39,34 +60,126 @@ export default {
     },
     data () {
         return {
+            // String-based filters applied to the list
             filters: "",
+            // The known recipes in the cookbook
             recipes: [],
-            keywords: [],
+            // All keyword to filter the recipes for (conjunctively)
             keywordFilter: [],
         }
     },
     computed: {
-        filteredRecipes () {
-            if (!this.filters) {
-                return this.recipes
-            } else if (this.recipes.length) {
-                let filtered = []
-                for (let i=0; i<this.recipes.length; i++) {
-                    if (this.recipes[i].name.toLowerCase().indexOf(this.filters.toLowerCase()) >= 0) {
-                        filtered.push(this.recipes[i])
+        // An array of all keywords in all recipes. These are neither sorted nor unique
+        rawKeywords() {
+            var keywordArray = this.recipes.map(function(r){
+                if(! 'keywords' in r) {
+                    return []
+                }
+                if(r.keywords != null){
+                    return r.keywords.split(',')
+                } else {
+                    return []
+                }
+            })
+            return [].concat(... keywordArray)
+        },
+        // An array of sorted and unique keywords over all the recipes
+        uniqKeywords() {
+            function uniqFilter(value, index, self) {
+                return self.indexOf(value) === index
+            }
+            return this.rawKeywords.sort().filter(uniqFilter)
+        },
+        // An array of objects that contain the keywords plus a count of recipes asociated with these keywords
+        keywordsWithCount() {
+            let $this = this
+            return this.uniqKeywords.map(function (kw){
+                return {
+                    'name': kw,
+                    'count': $this.rawKeywords.filter((kw2) => kw === kw2).length
+                }
+            }).sort(function (k1, k2) {
+                if(k1.count != k2.count) {
+                    // Distinguish by number
+                    return k2.count - k1.count
+                }
+                // Distinguish by keyword name
+                return (k1.name.toLowerCase() > k2.name.toLowerCase()) ? 1 : -1
+            })
+        },
+        // An array of keyword objects that are currently in use for filtering
+        selectedKeywords() {
+            return this.keywordsWithCount.filter((kw) => this.keywordFilter.includes(kw.name))
+        },
+        // An array of those keyword objects that are currently not in use for filtering
+        unselectedKeywords() {
+            return this.keywordsWithCount.filter((kw) => ! this.selectedKeywords.includes(kw))
```

Comment:
Shouldn’t it be faster to do the same (inverse) as for `selectedKeywords`?

```
return this.keywordsWithCount.filter((kw) => ! this.keywordFilter.includes(kw.name))
```

Reply:
This might be a small bit faster to use the `keywordFilter` instead of `selectedKeywords`. The difference is the `.name` access. Both `keywordFilter` and `selectedKeywords` should have the same number of entries though.

Which one is better readable?

Reply:
let’s say with your code it’s obvious that no keyword is missing since one partial list is generated by excluding entries which are present in the other partial list

Reply:
Just tell me what we should do. I'd like to finish this issue soon :smile:.

Reply:
lol. leave it like this^^